### PR TITLE
No node selectors and separate services

### PR DESCRIPTION
**Pull request by fabiand**
This is an alternate PR to #538.

Instead of using ingress, separate services are used to grant inbound access.

### CODE REVIEW

File: cluster/vagrant/setup_kubernetes_master.sh [hunk 1]
```diff
@@ -50,6 +50,31 @@ else
   kubectl create -f kube-$NETWORK_PROVIDER.yaml
 fi
 
+{
+  pushd /vagrant
+  source hack/config.sh
+  popd
+
+  for SVC in spice-proxy:3128 virt-controller:8182 virt-api:8182 haproxy:8184 virt-manifest:8186;
+  do
```
Comment (davidvossel) [inline cluster/vagrant/setup_kubernetes_master.sh]: I'd rather see this as part of 'make vagrant-deploy' than being performed during the initial 'vagrant up'. 

My concern is that changes to ingress shouldn't require someone to rebuild vagrant.

File: manifests/haproxy.yaml.in [hunk 2]
```diff
@@ -44,5 +42,3 @@ spec:
           periodSeconds: 20
       securityContext:
         runAsNonRoot: true
-      nodeSelector:
```
Comment (davidvossel) [inline manifests/haproxy.yaml.in]: I was under the impression something about our haproxy multiplexing between the k8s api server and virt-api required special node placement. 

From looking at this in more detail, the haproxy config appears to be referencing DNS names instead of hard coded IPs, so it seems like the node selector wouldn't be necessary. 

Does removing the node select actually work with ci?

File: tests/vm_migration_test.go [hunk 3]
```diff
@@ -45,7 +45,7 @@ var _ = Describe("VmMigration", func() {
 
 	var sourceVM *v1.VirtualMachine
 
-	var TIMEOUT float64 = 10.0
```
Comment (davidvossel) [inline tests/vm_migration_test.go]: Are these timeout increases related to this patch series?  I'm just curious if something about these changes introduce more latency into our functional tests.  It doesn't make sense to me how that would be possible.

File: manifests/iscsi-auth-demo-target.yaml.in [hunk 4]
```diff
@@ -50,5 +50,3 @@ spec:
         - name: host
           hostPath:
             path: /
-      nodeSelector:
```
Comment (davidvossel) [inline manifests/iscsi-auth-demo-target.yaml.in]: It's strange to me we had a nodeselector to begin with for the iscsi demo.

File: cluster/vagrant/setup_kubernetes_master.sh [hunk 5]
```diff
@@ -50,6 +50,31 @@ else
   kubectl create -f kube-$NETWORK_PROVIDER.yaml
 fi
 
+{
+  pushd /vagrant
+  source hack/config.sh
+  popd
+
+  for SVC in spice-proxy:3128 virt-controller:8182 virt-api:8182 haproxy:8184 virt-manifest:8186;
```
Comment (davidvossel) [inline cluster/vagrant/setup_kubernetes_master.sh]: Do we really need a virt-controller service endpoint?  It seems like we could just remove both the external and internal virt-controller endpoints.

**Comment by fabiand** [comment]
Single node, single master:
```
Ran 47 of 47 Specs in 879.951 seconds
SUCCESS! -- 47 Passed | 0 Failed | 0 Pending | 0 Skipped PASS
```

File: cluster/vagrant/setup_kubernetes_master.sh (hunk 5 above)
Comment (fabiand) [reply cluster/vagrant/setup_kubernetes_master.sh]: I just keep compatibility with the current state. Happy to drop all unnecessary mappings in a separate patch.

File: manifests/haproxy.yaml.in (hunk 2 above)
Comment (fabiand) [reply manifests/haproxy.yaml.in]: Seems so, see result below.

File: tests/vm_migration_test.go (hunk 3 above)
Comment (fabiand) [reply tests/vm_migration_test.go]: No, it's independent of these changes. It's reuiqred on my host to make the tests pass at all.

File: cluster/vagrant/setup_kubernetes_master.sh (hunk 1 above)
Comment (fabiand) [reply cluster/vagrant/setup_kubernetes_master.sh]: So ... We can do it there as well, obviously.

One of the things I'm trying to achieve is a spearation of the internal setup from the external access.
With this patch set this is achieved, because `manifests/` only contains internal pod definitions.
And ingress (even if not an ingress object) is left to the env setup - that's why I moved it to the vagrant setup.

In this setup, I can take the `manifests/` directory and use it for minikube, and adding a note that the user needs to run `kubectl expose service spice-proxy …` to get spice/console access.
What I want to say: The `manifest/` manifests are sufficient and independent to define KubeVirt in the cluster.

By moving the external services into `manifests/` we are pulling back env dependent configs into the manifest dir.

Comment (davidvossel) [reply cluster/vagrant/setup_kubernetes_master.sh]: @fabiand how about moving the ingress service generation code to cluster/deploy.sh then?

Comment (fabiand) [reply cluster/vagrant/setup_kubernetes_master.sh]: Great idea!

**Comment by fabiand** [comment]
Two nodes, one master:

```
Ran 47 of 47 Specs in 914.519 seconds
SUCCESS! -- 47 Passed | 0 Failed | 0 Pending | 0 Skipped PASS
```

Now I'll move the code as requested further up.

**Comment by fabiand** [comment]
retest this please

**Comment by fabiand** [comment]
@davidvossel the logic has moved to the deployment script. It should now also run on the CI.

**Comment by fabiand** [comment]
And a re-run:

```
Waiting for namespace kubevirt-test-default to be removed, this can take a while ...
Waiting for namespace kubevirt-test-alternative to be removed, this can take a while ...

Ran 48 of 48 Specs in 958.986 seconds
SUCCESS! -- 48 Passed | 0 Failed | 0 Pending | 0 Skipped PASS
ok  	kubevirt.io/kubevirt/tests	959.252s
[root@host-02 kubevirt]# k get nodes
kubevirt
NAME      STATUS    ROLES     AGE       VERSION
master    Ready     master    5h        v1.8.2
node0     Ready     <none>    5h        v1.8.2
node1     Ready     <none>    5h        v1.8.2
```

Looks safe to test.

**Comment by fabiand** [comment]
retest this please

**Comment by fabiand** [comment]
ok to test

**Comment by fabiand** [comment]
retest this please

**Comment by fabiand** [comment]
ci retest this please

**Comment by fabiand** [comment]
ci test please

**Comment by fabiand** [comment]
@rmohr I just see something in 282:
```                                // Make sure we schedule the VM to master       
                                vm.Spec.NodeSelector = map[string]string{"kubernetes.io/hostname": primaryNodeName}
```

But this is about pinning a VM.

**Comment by rmohr** [comment]
> But this is about pinning a VM.

You are right. It would just remove the dependency of our test  to the node name, but you did not remove that part.
